From f028513101dbcfb4b315abfc18a478dda35ac6d8 Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Wed, 17 Jun 2020 10:57:15 +0200
Subject: [PATCH] Fedora build patches.

---
 src/Makefile | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index b79a59ac5..be4b35096 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -43,7 +43,7 @@ endif
 ## Makefile flags
 ##
 
-MAKEFLAGS               += -l -j 8 -rR --no-print-directory
+MAKEFLAGS               += -l -rR --no-print-directory
 
 ifneq ($(findstring clean,$(MAKECMDGOALS)),)
 MAKEFLAGS               += -j 1
@@ -222,10 +222,7 @@ endif
 endif
 
 ifeq ($(DEBUG),0)
-CFLAGS                  += -O2
-ifneq ($(UNAME),Darwin)
-LFLAGS                  += -s
-endif
+CFLAGS                  += -DNDEBUG
 else
 ifeq ($(DEBUG),1)
 ifneq ($(UNAME),Darwin)
@@ -461,7 +458,7 @@ endif
 
 .PHONY: install
 ifeq ($(SHARED),1)
-install: install_docs install_shared install_library install_library_dev install_kernels install_modules install_hashcat
+install: install_shared install_library install_library_dev install_kernels install_modules install_hashcat
 else
 install: install_docs install_shared                                     install_kernels install_modules install_hashcat
 endif
@@ -527,7 +524,7 @@ install_kernels: install_shared
 install_modules: install_shared modules
 	$(INSTALL) -m 755 -d                                                    $(DESTDIR)$(SHARED_FOLDER)/modules
 	$(FIND) modules/  -mindepth 1 -type d -execdir $(INSTALL) -m 755 -d     $(DESTDIR)$(SHARED_FOLDER)/modules/{} \;
-	$(FIND) modules/  -mindepth 1 -type f -execdir $(INSTALL) -m 644 {}     $(DESTDIR)$(SHARED_FOLDER)/modules/{} \;
+	$(FIND) modules/  -mindepth 1 -type f -execdir $(INSTALL) -m 755 {}     $(DESTDIR)$(SHARED_FOLDER)/modules/{} \;
 
 .PHONY: install_library
 install_library: $(HASHCAT_LIBRARY)
-- 
2.31.1

