From ba735dbda88f0881b5193eeebc731bd2065202af Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Wed, 17 Jun 2020 10:57:15 +0200
Subject: [PATCH 1/2] Fedora build patches.

---
 src/Makefile | 19 ++++++++-----------
 1 file changed, 8 insertions(+), 11 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index af29457c..f5799a75 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -3,16 +3,16 @@
 ## License.....: MIT
 ##
 
-SHARED                  := 0
+SHARED                  := 1
 DEBUG                   := 0
 PRODUCTION              := 1
 PRODUCTION_VERSION      := v6.0.0
 ENABLE_CUBIN            := 1
 ENABLE_BRAIN            := 1
 USE_SYSTEM_LZMA         := 0
-USE_SYSTEM_ZLIB         := 0
-USE_SYSTEM_OPENCL       := 0
-USE_SYSTEM_XXHASH       := 0
+USE_SYSTEM_ZLIB         ?= 1
+USE_SYSTEM_OPENCL       ?= 1
+USE_SYSTEM_XXHASH       ?= 1
 
 ##
 ## Detect Operating System
@@ -40,7 +40,7 @@ endif
 ## Makefile flags
 ##
 
-MAKEFLAGS               += -l -j 8 -rR --no-print-directory
+MAKEFLAGS               += -l -rR --no-print-directory
 
 ifneq ($(findstring clean,$(MAKECMDGOALS)),)
 MAKEFLAGS               += -j 1
@@ -186,10 +186,7 @@ CFLAGS_ZLIB             += -DIOAPI_NO_64
 endif
 
 ifeq ($(DEBUG),0)
-CFLAGS                  += -O2
-ifneq ($(UNAME),Darwin)
-LFLAGS                  += -s
-endif
+CFLAGS                  += -DNDEBUG
 else
 ifeq ($(DEBUG),1)
 ifneq ($(UNAME),Darwin)
@@ -391,7 +388,7 @@ Only Linux, FreeBSD and Darwin can use the 'install' target
 endif
 
 ifeq ($(SHARED),1)
-install: install_docs install_shared install_library install_library_dev install_kernels install_modules install_hashcat
+install: install_shared install_library install_library_dev install_kernels install_modules install_hashcat
 else
 install: install_docs install_shared                                     install_kernels install_modules install_hashcat
 endif
@@ -451,7 +448,7 @@ install_kernels: install_shared
 install_modules: install_shared modules
 	$(INSTALL) -m 755 -d                                                    $(DESTDIR)$(SHARED_FOLDER)/modules
 	$(FIND) modules/  -mindepth 1 -type d -execdir $(INSTALL) -m 755 -d     $(DESTDIR)$(SHARED_FOLDER)/modules/{} \;
-	$(FIND) modules/  -mindepth 1 -type f -execdir $(INSTALL) -m 644 {}     $(DESTDIR)$(SHARED_FOLDER)/modules/{} \;
+	$(FIND) modules/  -mindepth 1 -type f -execdir $(INSTALL) -m 755 {}     $(DESTDIR)$(SHARED_FOLDER)/modules/{} \;
 
 install_library: $(HASHCAT_LIBRARY)
 	$(INSTALL) -m 755 -d                                                    $(DESTDIR)$(LIBRARY_FOLDER)
-- 
2.26.2

